File: server/internal/model/moment.go
/*
项目名称：JeriBlog
文件名称：moment.go
创建时间：2026-04-16 15:00:36

系统用户：Jerion
作　　者：Jerion
联系邮箱：416685476@qq.com
功能描述：动态数据模型
*/

package model

import (
	"time"
)

// Moment 动态模型
type Moment struct {
	ID          uint       `gorm:"primarykey" json:"id"`
	Content     string     `gorm:"type:json;not null" json:"content"` // 内容（JSON）- 包含text、images、location、link、music、video等
	IsPublish   bool       `gorm:"default:true" json:"is_publish"`    // 是否发布
	PublishTime *time.Time `json:"publish_time"`                      // 发布时间
	CreatedAt   time.Time  `json:"created_at"`
	UpdatedAt   time.Time  `json:"updated_at"`
}

File: blog/types/moment.ts
/**
 * 视频内容
 */
export interface MomentVideo {
  url: string
  platform?: 'youtube' | 'bilibili' | 'local'
  video_id?: string
}

/**
 * 音乐内容
 */
export interface MomentMusic {
  server: 'netease' | 'tencent' | 'kugou' | 'xiami' | 'baidu'
  type: 'song' | 'playlist' | 'album' | 'search' | 'artist'
  id: string
}

/**
 * 链接内容
 */
export interface MomentLink {
  url: string
  title: string
  favicon?: string
}

/**
 * 动态内容
 */
export interface MomentContent {
  text?: string
  images?: string[]
  location?: string
  tags?: string
  video?: MomentVideo
  music?: MomentMusic
  link?: MomentLink
}

/**
 * 动态
 */
export interface Moment {
  id: number
  content: MomentContent
  publish_time: string
}

/**
 * 动态列表响应
 */
export interface MomentListResponse {
  list: Moment[]
  total: number
  page: number
  page_size: number
}

